app/controllers/admin/mentors_controller.rb

```diff
@@ -1,4 +1,18 @@
 class Admin::MentorsController < ApplicationController
+
   def index
+    @user_ids = TrackMentorship.distinct.pluck(:user_id)
+    @users = User.where(id: @user_ids)
+    @tracks = Hash.new
+    @user_ids.each do |user_id|
+      @tracks[user_id] = TrackMentorship.where(user_id: user_id).to_a().map {
```

Comment:
There's a couple of issues with this block.

1) It's currently doing an "n+1 query" (have fun on Google). You can avoid this by using `.includes(:track)`
2) You don't need `to_a()`
3) You can avoid the map using a `join` and `pluck`. So `joins(:track).pluck("tracks.title")`